eessi_bot_event_handler.py

```diff
@@ -599,6 +601,61 @@ def start(self, app, port=3000):
         self.log(log_file_info)
         waitress.serve(app, listen='*:%s' % port)
 
+    def handle_pull_request_merged_event(self, event_info, pr):
+        """
+        Handle events of type pull_request with the action merged. Main action
+        is to scan directories used and move them to the trash_bin.
+
+        Args:
+        event_info (dict): event received by event_handler
+        pr (github.PullRequest.PullRequest): instance representing the pull request
+
+        Returns:
+        github.IssueComment.IssueComment instance or None (note, github refers to
+        PyGithub, not the github from the internal connections module)
+        """
+
+        # Detect event and only act if PR is merged
+        request_body = event_info['raw_request_body']
+        action = request_body['action']
+
+        if action == 'merged':
+            self.log("PR merged: scanning directories used by PR")
+            self.log(f"merge '{action}' action is handled")
+        else:
+            self.log(f"merge action '{action}' not handled")
+            return
+        # at this point we know that we are handling a new merge
+        # NOTE: Permissions to merge are already handled through GitHub, we
+        # don't need to check here
+
+        clean_up_comments_cfg = self.cfg[config.SECTION_CLEAN_UP_COMMENTS]
+        # 1) determine the jobs that have been run for the PR
+        job_dirs = determine_job_dirs(pr.number)
+
+        # 2) Get trash_bin_dir from configs
+        trash_bin_root_dir = self.cfg[config.SECTION_MERGED_PR][config.MERGED_PR_SETTING_TRASH_BIN_ROOT_DIR]
+        repo_cfg = get_repo_cfg(self.cfg)
+        repo_name = repo_cfg[cvmfs_repository.REPOS_CFG_REPO_NAME]
+        dt = datetime.now(timezone.utc)
+        trash_bin_dir = "/".join([trash_bin_root_dir, repo_name, dt.strftime('%Y%m%d')])
```

Comment:
I'm a bit unsure whether we want to use the CVMFS repository or rather the target repository to which the merged/closed PR was made (e.g., EESSI/software-layer). The target repository should be available from the `event_info` dictionary.

Reply:
Handled in 769daec68918d09ee586dfac4795c0cc7719bced and 4160bd1